FanController.cpp

```diff
@@ -0,0 +1,62 @@
+#include "FanController.hpp"
+
+class FanController {
+   private:
+    map<FanMode, string> modeNames = {
+        {FanMode::OFF, "OFF"},
+        {FanMode::LOW, "LOW"},
+        {FanMode::MEDIUM, "MEDIUM"},
+        {FanMode::HIGH, "HIGH"},
+        {FanMode::AUTO, "AUTO"}};
+
+    FanMode currentMode = FanMode::OFF;
+
+   public:
+    FanController(FanMode starting_mode = FanMode::OFF) {
+        currentMode = starting_mode;
+    }
+
+    void setMode(FanMode new_mode) {
+        currentMode = new_mode;
+    }
+
+    void nextMode() {
```

Comment:
Način na koji je metoda nextMode() implementirana nije optimalan. Može se riješiti prateći istu logiku kao u metodi previousMode()